web_client/dialogs/openAnnotatedImage.js

```diff
@@ -1,57 +1,134 @@
+        this.$el.html(listTemplate({
+            items: this.collection.toJSON(),
+            paths
+        }));
+        return this;
+    }
+});
 
 const OpenAnnotatedImage = View.extend({
     events: {
-        'click .h-annotated-image': '_submit'
+        'click .h-annotated-image': '_submit',
+        'keyup input': '_updateQuery',
+        'change select': '_updateQuery'
     },
 
     initialize() {
         this.collection = new ItemCollection();
         // disable automatic sorting of this collection
         this.collection.comparator = null;
-        this.listenTo(this.collection, 'reset', this.render);
+
+        this._query = new backbone.Model({
+            imageName: '',
+            creator: ''
+        });
+        this._nextQuery = {};
+
+        this._users = new UserCollection();
+        this._users.sortField = 'login';
+        this._users.pageLimit = 500;
+        this._usersIsFetched = false;
+        this._users.fetch().done(() => {
+            this._usersIsFetched = true;
+            this.render();
+            this._fetchImages();
+        });
+        this.listenTo(this._query, 'change', this._queueFetchImages);
     },
 
     render() {
+        if (!this._usersIsFetched) {
+            return this;
+        }
         this.$el.html(template({
-            items: this.collection.toJSON(),
-            paths
+            imageName: this._query.get('imageName'),
+            creator: this._query.get('creator'),
+            users: this._users
         })).girderModal(this);
         this.$el.tooltip();
+
+        new AnnotatedImageList({
+            parentView: this,
+            collection: this.collection,
+            el: this.$('.h-annotated-images-list-container')
+        }).render();
         return this;
     },
 
-    fetch() {
-        var items;
-        return restRequest({
+    _fetchImages() {
+        const data = this._nextQuery;
+        let items;
+
+        if (!this._nextQuery) {
+            return;
+        }
+        delete this._nextQuery;
+
+        data.limit = 10;
+        restRequest({
             url: 'annotation/images',
-            data: {
-                limit: 10
-            }
+            data
         }).then((_items) => {
             items = _items;
             const promises = _.map(items, (item) => {
                 return this._getResourcePath(item);
             });
             return $.when(...promises);
-        }).then(() => {
+        }).done(() => {
             this.collection.reset(items);
-            return items;
+            this._fetchImages();
+        });
+    },
+
+    _queueFetchImages() {
+        const imageName = this._query.get('imageName');
+        const creator = this._query.get('creator');
+        this._nextQuery = {};
+
+        if (imageName) {
+            this._nextQuery.imageName = imageName;
+        }
+        if (creator) {
+            this._nextQuery.creatorId = creator;
+        }
+
+        this._fetchImages();
```

Comment:
This will always call `_fetchImages` when there is an input change.  To prevent having multiple calls at once, you probably need to add a value like `this._inFetch = true;` just before the rest request that is cleared when the `done` function is called, and then `_fetchImages` would not get called here if `this._inFetch` was set.

Reply:
That's what I was attempting with the `this._nextQuery`, but I see now my logic was flawed.  I'll make another attempt.